### PR TITLE
Hospital database pull request

### PR DESCRIPTION
**Pull request by Solobarine**
This pull request is to seek a review and approval of the database created from the diagram given.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+DROP TABLE IF EXISTS invoices CASCADE;
+DROP TABLE IF EXISTS invoice_items;
+
+-- Create the main tables
+CREATE TABLE medical_histories ( id INT GENERATED ALWAYS AS IDENTITY, admitted_at DATE, patient_id INT, status varchar(30), PRIMARY KEY(id));
+CREATE TABLE patients ( id INT GENERATED ALWAYS AS IDENTITY, name VARCHAR(100), date_of_birth DATE, PRIMARY KEY(id));
+CREATE TABLE invoices ( id INT GENERATED ALWAYS AS IDENTITY, total_amount FLOAT, generated_at DATE, payed_at DATE, medical_history_id INT, PRIMARY KEY (id));
+CREATE TABLE invoice_items (id INT GENERATED ALWAYS AS IDENTITY, unit_price FLOAT, quantity INT, total_price FLOAT, invoice_id INT, treatment_id INT, PRIMARY KEY (id));
+CREATE TABLE treatments (id INT GENERATED ALWAYS AS IDENTITY, type VARCHAR(60), name VARCHAR(80), PRIMARY KEY(id));
+CREATE TABLE medical_histories_treatments( treatments_id INT, medical_histories_id INT, PRIMARY KEY(treatments_id,medical_hitories_id));
```
Comment (Davidosky007) [inline schema_based_on_diagram.sql]: * Kindly stick to the types shown in the diagram ( timestamp, decimal)

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+-- ALTER TABLES TO ADD FOREIGN KEYS
+ALTER TABLE invoices ADD CONSTRAINT fk_medical_history_id FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id);
+ALTER TABLE invoice_items ADD CONSTRAINT fk_invoice_id FOREIGN KEY (invoice_id) REFERENCES invoices(id);
+ALTER TABLE invoice_items ADD CONSTRAINT fk_treatment_id FOREIGN KEY (treatment_id) REFERENCES treatments(id);
+ALTER TABLE medical_histories ADD CONSTRAINT fk_patient_id FOREIGN KEY (patient_id) REFERENCES patients(id);
+ALTER TABLE medical_histories_treatments FOREIGN KEY (treatments_id) REFERENCES treatments(id);
+ALTER TABLE medical_histories_treatments FOREIGN KEY (medical_histories_id) REFERENCES medical_histories(id);
```
Comment (Davidosky007) [inline schema_based_on_diagram.sql]: * Ensure that all FOREIGN KEYS have indexes created, [Check here for reference ](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md#:~:text=Remember%20to%20add%20the%20FK%20indexes.)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Solobarine @eng-mohamed-eldeeb 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* Proper use of Gitflow 👍 
* Good looking readme file 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        ON DELETE CASCADE,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE treatments (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  type VARCHAR(50) NOT NULL,
+  name VARCHAR(50) NOT NULL,
+  PRIMARY KEY (id)
+);
```
Comment (DeliceLydia) [inline schema_based_on_diagram.sql:47]: - [ ] Please add indexes to all foreign keys
like:
```
CREATE INDEX ON medical_histories (patient_id);
CREATE INDEX ON invoices (medical_history_id);
CREATE INDEX ON invoice_items (invoice_id);
CREATE INDEX ON invoice_items (treatment_id);
CREATE INDEX ON medical_histories_has_treatments (medical_history_id);
CREATE INDEX ON medical_histories_has_treatments (treatment_id);
```

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._


### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by mugishasam123 — APPROVED** [review]
Hi @Solobarine  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors 👍 
- Descriptive PR title 🥇 

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**